frontend/src/DialectReportView.tsx

```diff
@@ -22,27 +28,82 @@ export const DialectReportView = ({
   }, [reportData]);
 
   const filteredData = useMemo(() => {
-    let filteredData = reportData;
-    if (params.getAll("language").length) {
-      const filteredArray = Array.from(
+    const filteredReportData = { ...reportData };
+    let filteredOtherImplementationsData: Record<string, Implementation> = {};
+    const selectedLanguages = params.getAll("language");
+
+    if (selectedLanguages.length > 0) {
+      const filteredReportArray = Array.from(
         reportData.implementations.entries(),
       ).filter(([, data]) =>
-        params.getAll("language").includes(data.metadata.language),
+        selectedLanguages.includes(data.metadata.language),
+      );
+      const filteredReportImplementationsMap = new Map(filteredReportArray);
+      filteredReportData.implementations = filteredReportImplementationsMap;
+      if (allImplementationsData) {
```

Comment:
Hard to examine all the possible scenarios - can `allImplementationsData` be falsy?

Reply:
Yes @harrel56  actually on the `DragAndDrop` component when the `DialectReportView` gets rendered we don't have the `allImplementationsData`  over there so I made it optional.

Reply:
Oh right, now I see it. So maybe it's better to make it required and pass empty object in `DragAndDrop`? That way you can get rid of the ifs and it would simplify it a bit. But that's just a suggestion

Reply:
No problem I will do that